File: ios/exe/Packages/ExeDomain/Sources/ExeDomain/DeviceToken.swift
import Foundation

public enum DeviceTokenKind: String, Codable, CaseIterable, Sendable {
    case fcm
    case voip
}

public struct DeviceToken: Codable, Hashable, Identifiable, Sendable {
    public let createdAt: DateTime
    public let environment: ExeEnvironment
    public let id: String
    public let kind: DeviceTokenKind
    public let platform: String
    public let token: String
    public let updatedAt: DateTime
    public let userId: UserID
}

File: app/packages/domain/src/device-token.ts
import { dateTimeSchema, environmentSchema, userIdSchema } from './common';
import { z } from 'zod';

export const deviceTokenKindSchema = z.enum(['fcm', 'voip']);

export const deviceTokenSchema = z
  .object({
    createdAt: dateTimeSchema,
    environment: environmentSchema,
    id: z.string().min(1),
    kind: deviceTokenKindSchema,
    platform: z.literal('ios'),
    token: z.string().min(1),
    updatedAt: dateTimeSchema,
    userId: userIdSchema,
  })
  .strict();

export type DeviceToken = z.infer<typeof deviceTokenSchema>;

export type DeviceTokenKind = z.infer<typeof deviceTokenKindSchema>;
